### PR TITLE
The Camping Update

### PR DESCRIPTION
**Pull request by musicmanvr**
yeehaw

## About the PR

Adds foldable bedrolls and craftable campfires.

Bedrolls can be gotten from the surplus uniform vendor or from cargo's uniform vendor.
Campfires cost 4 wood to build and take 5 seconds to build.
## Why / Balance

FoB can get really boring, and I saw a video on youtube the other day that showed a poor dude lying on a stretcher bed instead of a camping bedroll.

It allows for more immersion within marine squads taking a brief camp in the field.
Bedrolls are slower to pull than rollerbeds, however, so they aren't just a direct upgrade.
Bedrolls also provide the same tiny healing bonus as regular beds.
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image

image

## Breaking changes

**Changelog**

:cl:
- add: Added bedrolls and campfires for camping in the field.

### CODE REVIEW

File: Resources/Prototypes/_RMC14/Recipes/Construction/structures.yml
Comment (musicmanvr) [inline Resources/Prototypes/_RMC14/Recipes/Construction/structures.yml:1]: no idea what the phantom space removal is about but I didn't touch it

File: Resources/Prototypes/_RMC14/Entities/Structures/Furniture/beds.yml [hunk 1]
```diff
@@ -26,3 +26,100 @@
...
+  parent: [BaseItem, BaseFoldable]
+  id: Bedroll
+  name: bedroll
+  description: A foldable bedroll, just about the only thing of comfort in the field. You're generally supposed to unroll it before sleeping, but who needs rules?
+  components:
+  - type: HealOnBuckle
+    damage:
+      types:
+        Poison: -0.1
+        Blunt: -0.1
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Structures/Furniture/beds.yml:42]:
```suggestion
```

File: Resources/Prototypes/_RMC14/Entities/Structures/Furniture/beds.yml [hunk 2]
```diff
@@ -26,3 +26,100 @@
...
+    - state: bedroll_folded
+      map: ["foldedLayer"]
+      visible: false
+    - state: bedroll_pillow
+      map: ["buckledLayer"]
+      visible: false
+  - type: MovedByPressure
+  - type: DamageOnHighSpeedImpact
+    soundHit:
+      collection: WeakHit
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Structures/Furniture/beds.yml:62]:
```suggestion
```

File: Resources/Prototypes/_RMC14/Entities/Structures/Furniture/beds.yml [hunk 3]
```diff
@@ -26,3 +26,100 @@
...
+          damage: 50
+        behaviors:
+          - !type:PlaySoundBehavior
+            sound:
+              path: /Audio/Effects/poster_broken.ogg
+          - !type:SpawnEntitiesBehavior
+            spawn:
+              MaterialCloth1:
+                min: 1
+                max: 2
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Structures/Furniture/beds.yml:88]:
```suggestion
```

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]
Looks good just a few removals

**Comment by musicmanvr** [comment]
> Looks good just a few removals

fine with these

File: Resources/Textures/_RMC14/Structures/campfire.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    0.3,
+    0.3,
+    0.3,
+    0.3,
+    0.3
+    ]
+    ]
+    }
+  ]
+}
```
Comment (DrSmugleaf) [inline Resources/Textures/_RMC14/Structures/campfire.rsi/meta.json:26]: I am baffled at how cooked the indentation got here
```suggestion
{
  "version": 1,
  "license": "CC-BY-SA-3.0",
  "copyright": "https://github.com/cmss13-devs/cmss13/blob/7cb618c69b75873f3ce893022fe08d1233b3152d/icons/obj/structures/structures.dmi",
  "size": {
    "x": 32,
    "y": 32
  },
  "states": [
    {
      "name": "campfire"
    },
    {
      "name": "burning",
      "delays": [
        [
          0.3,
          0.3,
          0.3,
          0.3,
          0.3
        ]
      ]
    }
  ]
}
```

**Review by DrSmugleaf — DISMISSED** [review]

File: Resources/Textures/_RMC14/Structures/campfire.rsi/meta.json (hunk 4 above)
Comment (musicmanvr) [reply Resources/Textures/_RMC14/Structures/campfire.rsi/meta.json]: no idea what happened here

**Review by QuietlyWhisper — APPROVED** [review]

**Pull request merged**